reconcile/utils/gql.py

```diff
@@ -229,26 +229,38 @@ def get_git_commit_info(sha, server, token=None):
 
 @retry(exceptions=requests.exceptions.ConnectionError, max_attempts=5)
 def init_from_config(
-    sha_url=True, integration=None, validate_schemas=False, print_url=True
+    autodetect_sha=True,
+    sha=None,
+    integration=None,
+    validate_schemas=False,
+    print_url=True,
 ):
+    server, token = _get_gql_server_and_token(autodetect_sha=autodetect_sha, sha=sha)
+
+    if print_url:
+        logging.info(f"using gql endpoint {server}")
+    return init(server, token, integration, validate_schemas)
+
+
+def _get_gql_server_and_token(
+    autodetect_sha: bool = False, sha: Optional[str] = None
+) -> Tuple[str, str]:
     config = get_config()
 
     server_url = urlparse(config["graphql"]["server"])
     server = server_url.geturl()
-
     token = config["graphql"].get("token")
-    if sha_url:
+    if sha:
+        server = server_url._replace(path=f"/graphqlsha/{sha}").geturl()
+    elif autodetect_sha:
         sha = get_sha(server_url, token)
         server = server_url._replace(path=f"/graphqlsha/{sha}").geturl()
-
+    if sha:
         runing_state = RunningState()
```

Comment:
```suggestion
        running_state = RunningState()
```

Reply:
one would say this is unrelated to the changes in the PR :sweat_smile:

Reply:
[f6aedd9](https://github.com/app-sre/qontract-reconcile/pull/2672/commits/f6aedd9113e00d0faa506d6c6a2f34087a3a7d59)